### ISSUE TITLE
`ClassCastException` after upgrading to v7.0.0

### ISSUE THREAD

**Issue by alvr** [getsentry/sentry-java#3103]
### Integration

sentry-android

### Build System

Gradle

### AGP Version

8.2.0

### Proguard

Enabled

### Version

7.0.0

### Steps to Reproduce

After upgrading to v7.0.0 from v6.34.0 on Android, it crashes on start with R8 fullMode enabled.

Although I use Kotlin Multiplatfrom with Sentry KMP, there is no problem with the previous version.

I don't use Sentry NDK and in the _External libraries_ dropdown of Android Studio it doesn't appear.

In debug mode the app works, only crashes in release mode.

Here is a reproducible project: https://github.com/alvr/katana/tree/renovate/major-sentry

### Expected Result

Just works as the previous version of Sentry.

### Actual Result

Crash on start.

```
java.lang.ClassCastException: io.sentry.android.ndk.NdkScopeObserver cannot be cast to io.sentry.cache.PersistingScopeObserver
    at io.sentry.Scope.addBreadcrumb
    at io.sentry.Hub.addBreadcrumb(SourceFile:379)
    at io.sentry.Hub.addBreadcrumb(SourceFile:385)
    at io.sentry.android.core.SentryAndroid.init
    at io.sentry.android.core.SentryAndroid.init
    at io.sentry.kotlin.multiplatform.SentryInit_androidKt.initSentry
    at io.sentry.kotlin.multiplatform.SentryBridge.init
    at io.sentry.kotlin.multiplatform.Sentry.init
    at dev.alvr.katana.ui.main.KatanaAppKt.initSentry
    at dev.alvr.katana.ui.main.KatanaAppKt.initApp
    at dev.alvr.katana.ui.main.KatanaAppKt.access$initApp
    at dev.alvr.katana.ui.main.ComposableSingletons$KatanaAppKt$lambda-3$1.invoke
    at dev.alvr.katana.ui.main.ComposableSingletons$KatanaAppKt$lambda-1$1.invoke$bridge(SourceFile:0)
    at dev.alvr.katana.ui.main.ComposableSingletons$KatanaAppKt$lambda-3$1.invoke
    at dev.alvr.katana.ui.main.ComposableSingletons$KatanaAppKt$lambda-1$1.invoke$bridge(SourceFile:0)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:108)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:35)
    at org.koin.compose.KoinApplicationKt$KoinApplication$1.invoke
    at androidx.compose.material.AlertDialogKt$AlertDialogContent$1$1$1$1$1.invoke$bridge(SourceFile:0)
    at org.koin.compose.KoinApplicationKt$KoinApplication$1.invoke
    at androidx.compose.material.AlertDialogKt$AlertDialogContent$1$1$1$1$1.invoke$bridge(SourceFile:0)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:108)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:35)
    at androidx.compose.material.AlertDialogKt$$InternalSyntheticOutline$319$6d2b6eb8dcb566ad383e3fcd484ffd4c6183203bfb1f88bf0ccb8c5755d1c9bf$0.m(SourceFile:1)
    at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(SourceFile:0)
    at org.koin.compose.KoinApplicationKt.KoinApplication(SourceFile:111)
    at dev.alvr.katana.ui.main.KatanaAppKt.KatanaDI(SourceFile:58)
    at dev.alvr.katana.ui.main.KatanaAppKt.KatanaApp(SourceFile:38)
    at dev.alvr.katana.ComposableSingletons$MainActivityKt$lambda-1$1.invoke(SourceFile:13)
    at dev.alvr.katana.ComposableSingletons$MainActivityKt$lambda-1$1.invoke(SourceFile:13)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:108)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:35)
    at androidx.compose.ui.platform.ComposeView.Content(SourceFile:428)
    at androidx.compose.ui.platform.AbstractComposeView$ensureCompositionCreated$1.invoke(SourceFile:252)
    at androidx.compose.ui.platform.AbstractComposeView$ensureCompositionCreated$1.invoke$bridge(SourceFile:0)
    at androidx.compose.ui.platform.AbstractComposeView$ensureCompositionCreated$1.invoke
    at androidx.compose.foundation.layout.WrapContentElement$Companion$height$1.invoke$bridge(SourceFile:0)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:108)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:35)
    at androidx.compose.material.AlertDialogKt$$InternalSyntheticOutline$319$6d2b6eb8dcb566ad383e3fcd484ffd4c6183203bfb1f88bf0ccb8c5755d1c9bf$0.m(SourceFile:1)
    at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(SourceFile:0)
    at androidx.compose.ui.platform.CompositionLocalsKt.ProvideCommonCompositionLocals(SourceFile:195)
    at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt$ProvideAndroidCompositionLocals$3.invoke
    at androidx.compose.foundation.lazy.layout.LazyLayoutPrefetcher_androidKt$LazyLayoutPrefetcher$2.invoke$bridge(SourceFile:0)
    at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt$ProvideAndroidCompositionLocals$3.invoke
    at androidx.compose.foundation.lazy.layout.LazyLayoutPrefetcher_androidKt$LazyLayoutPrefetcher$2.invoke$bridge(SourceFile:0)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:108)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:35)
    at androidx.compose.material.AlertDialogKt$$InternalSyntheticOutline$319$6d2b6eb8dcb566ad383e3fcd484ffd4c6183203bfb1f88bf0ccb8c5755d1c9bf$0.m(SourceFile:1)
    at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(SourceFile:0)
    at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt.ProvideAndroidCompositionLocals(SourceFile:110)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1$1$2.invoke(SourceFile:158)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1$1$2.invoke$bridge(SourceFile:0)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1$1$2.invoke(SourceFile:157)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1$1$2.invoke$bridge(SourceFile:0)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:108)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:35)
    at androidx.compose.material.AlertDialogKt$$InternalSyntheticOutline$319$6d2b6eb8dcb566ad383e3fcd484ffd4c6183203bfb1f88bf0ccb8c5755d1c9bf$0.m(SourceFile:1)
    at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(SourceFile:0)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1$1.invoke
    at androidx.compose.ui.platform.WrappedComposition$setContent$1$1$2.invoke$bridge(SourceFile:0)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1$1.invoke
    at androidx.compose.ui.platform.WrappedComposition$setContent$1$1$2.invoke$bridge(SourceFile:0)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:108)
    at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:35)
    at androidx.compose.runtime.ActualJvm_jvmKt.invokeComposable(SourceFile:78)
    at androidx.compose.runtime.ComposerImpl.doCompose(SourceFile:3340)
    at androidx.compose.runtime.ComposerImpl.composeContent$runtime_release(SourceFile:3273)
    at androidx.compose.runtime.CompositionImpl.composeContent(SourceFile:588)
    at androidx.compose.runtime.Recomposer.composeInitial$runtime_release(SourceFile:1013)
    at androidx.compose.runtime.CompositionImpl.setContent(SourceFile:520)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(SourceFile:142)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(SourceFile:133)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke$bridge(SourceFile:0)
    at androidx.compose.ui.platform.AndroidComposeView.setOnViewTreeOwnersAvailable(SourceFile:1191)
    at androidx.compose.ui.platform.WrappedComposition.setContent(SourceFile:133)
    at androidx.compose.ui.platform.WrappedComposition.onStateChanged(SourceFile:183)
    at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(SourceFile:314)
    at androidx.lifecycle.LifecycleRegistry.addObserver(SourceFile:192)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(SourceFile:140)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(SourceFile:133)
    at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke$bridge(SourceFile:0)
    at androidx.compose.ui.platform.AndroidComposeView.onAttachedToWindow(SourceFile:1266)
    at android.view.View.dispatchAttachedToWindow(View.java:21991)
    at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3493)
    at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3500)
    at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3500)
    at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3500)
    at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3500)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3198)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2650)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9526)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1343)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1352)
    at android.view.Choreographer.doCallbacks(Choreographer.java:952)
    at android.view.Choreographer.doFrame(Choreographer.java:882)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1326)
    at android.os.Handler.handleCallback(Handler.java:958)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:205)
    at android.os.Looper.loop(Looper.java:294)
    at android.app.ActivityThread.main(ActivityThread.java:8194)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

**Comment by romtsn** [comment]
Yes, as we mention in the [migration guide](https://docs.sentry.io/platforms/android/migration/#sentry-integrations-version-compatibility) you have to have all sentry versions aligned when bumping to `7.0.0`. Since the KMP sdk hasn't been shipped yet with Android 7.0.0, you'd either have to wait until it's out or force all the sentry version to 7.0.0 through Gradle (but not sure if everything will work either):
```
configurations.configureEach {
    resolutionStrategy.eachDependency {
        if (requested.group == "io.sentry") {
            useVersion("7.0.0")
        }
    }
}
```

**Comment by alvr** [comment]
Thanks, that's it! That is what happen when you don't read the documentation.

I used to avoid setting the version to the multiplatform library and using the catalog version.

```kotlin
configurations.configureEach {
    resolutionStrategy.eachDependency {
        if (requested.group == "io.sentry" && !requested.name.contains("multiplatform")) {
            useVersion(catalogVersion("sentry"))
        }
    }
}
```
